bin/apisix

```diff
@@ -798,333 +41,30 @@ version:    print the version of apisix
-            local post_json = '{"value":"' .. base64_encode("init_dir") ..  '", "key":"' .. base64_encode(key) .. '"}'
-            local cmd = "curl " .. uri .. " -X POST -d '" .. post_json
-                        .. "' --connect-timeout " .. timeout
-                        .. " --max-time " .. timeout * 2 .. " --retry 1 2>&1"
-
-            local res = execute_cmd(cmd)
-            if (etcd_version == "v3" and not res:find("OK", 1, true)) then
-                is_success = false
-                if (index == host_count) then
-                    error(cmd .. "\n" .. res)
-                end
-
-                break
-            end
-
-            if show_output then
-                print(cmd)
-                print(res)
-            end
-        end
-
-        if is_success then
-            etcd_ok = true
-            break
-        end
-    end
-
-    if not etcd_ok then
-        error("none of the configured etcd works well")
-    end
-end
-_M.init_etcd = init_etcd
-
-local openresty_args = [[openresty  -p ]] .. apisix_home .. [[ -c ]]
-                       .. apisix_home .. [[/conf/nginx.conf]]
-
-function _M.start(...)
-    -- check running
-    local pid_path = apisix_home .. "/logs/nginx.pid"
-    local pid, err = read_file(pid_path)
-    if pid then
-        local hd = io.popen("lsof -p " .. pid)
-        local res = hd:read("*a")
-        if res and res ~= "" then
-            print("APISIX is running...")
-            return nil
-        end
-    end
-
-    init(...)
-    init_etcd(...)
-
-    local cmd = openresty_args
-    -- print(cmd)
-    os.execute(cmd)
-end
-
-function _M.stop()
-    local cmd = openresty_args .. [[ -s stop]]
-    -- print(cmd)
-    os.execute(cmd)
-end
-
-function _M.restart()
-  _M.stop()
-  _M.start()
+local function version()
+    print(ver['VERSION'])
 end
 
-function _M.reload()
-    -- reinit nginx.conf
-    init()
-
-    local test_cmd = openresty_args .. [[ -t -q ]]
-    -- When success,
-    -- On linux, os.execute returns 0,
-    -- On macos, os.execute returns 3 values: true, exit, 0, and we need the first.
-    local test_ret = os.execute((test_cmd))
-    if (test_ret == 0 or test_ret == true) then
-        local cmd = openresty_args .. [[ -s reload]]
-        -- print(cmd)
-        os.execute(cmd)
-        return
-    end
-    print("test openresty failed")
-end
 
-function _M.version()
-    local ver = require("apisix.core.version")
-    print(ver['VERSION'])
-end
+local command = {
+    version = version,
+    help = help,
+    init = ops.init,
+    init_etcd = etcd.init,
+    start = ops.start,
+    stop = ops.stop,
+    restart = ops.restart,
+    reload = ops.reload,
+}
 
-local cmd_action = arg[1]
-if not cmd_action then
-    return _M.help()
+local action = arg[1]
+if not action then
+    return help()
 end
 
-if not _M[cmd_action] then
-    print("invalid argument: ", cmd_action, "\n")
+if not command[action] then
+    util.die("unknown sub-command: ", action, "\n")
```

Comment:
Please also integrate https://github.com/apache/apisix/commit/1fe4e50f5a153edc738efbe92aba3f910e5c888e into this refactor.

Reply:
OK, already did it.